molecule/eval/converge.yml

```diff
@@ -0,0 +1,113 @@
+---
+
+# EVAL FROM STRING
+
+- name: Eval function from string
+  hosts: cluster
+  roles:
+    - ansible-cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_app_name: 'myapp'
+    cartridge_scenario:
+      - eval
+    cartridge_eval_body: |
+        local fmt, age, t = ...
+
+        -- comment
+
+        --[[ one more comment ]]
+
+        ---[[ this is one-line comment
+        local argparse = require('cartridge.argparse')
+        local opts, err = argparse.get_opts({
+            app_name = 'string',
+            instance_name = 'string',
+        })
+        ---]]
+
+        --[[ ]] assert(err == nil, tostring(err))
+
+        local instance_id --[=[ ]=]
+        if opts.instance_name == nil then
+            instance_id = opts.app_name
+        else
+            instance_id = string.format('%s.%s', opts.app_name, opts.instance_name)
+        end
+
+        return unpack({
+            [[This snippet was evaled from string]],
+            string.format(fmt, instance_id),
+            string.format([=[I am %s seconds old]=], age),
+            require('yaml').encode(t),
+        })
+
+    cartridge_eval_args:
+      - Hello, my name is %s!
+      - 43
+      - {cute: table}
+
+- name: Print and check result of script from string
+  hosts: cluster
+  become: true
+  become_user: root
+  gather_facts: false
+  tasks:
+    - name: 'Debug result'
+      debug:
+        msg: '{{ eval_res }}'
+
+    - name: 'Check result'
+      assert:
+        msg: 'Received bad result'
+        success_msg: 'Received result is OK'
+        that: >-
+          eval_res == [
+              "This snippet was evaled from string",
+              "Hello, my name is " + instance_info.instance_id + "!",
+              "I am 43 seconds old",
+              "--- {'cute': 'table'}\n...\n",
+          ]
+
+## EVAL FROM FILE
```

Comment:
Let's swap the tests, so it will be tested that the body can be rewritten via vars after file eval use!

Reply:
As I see, you know all about such cases =)

Thanks!